From d66416bfdef2d36527e5bce2b24f8bd1ca515ee8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 30 Sep 2021 19:20:00 +0300
Subject: [PATCH] [SW-52828] boot_if: implement multi err/sts reg

Implement the multiple registers use for both the errors and device
status bits.
Update the existing internal API and update all users.

Change-Id: I215aaa74af6dabc1e3aa682053d6cfe04e426b6e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 886308a..8462b05 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -32,7 +32,8 @@
 				     struct boot_desc *desc, bool is_prime)
 {
 	bool is_valid = true;
-	u32 idx, bit;
+	u32 idx;
+	u64 boot_err;
 	char *msg;
 
 	idx = boot_desc_get_img_desc(image_pending, HL_FW_IMG_PREBOOT, desc);
@@ -58,13 +59,13 @@
 	}
 	if (!is_valid) {
 		if (is_prime) {
-			bit = CPU_BOOT_ERR0_PRI_IMG_VER_FAIL;
+			boot_err = BIT_ULL(CPU_BOOT_ERR_PRI_IMG_VER_FAIL);
 			msg = "Primary";
 		} else {
-			bit = CPU_BOOT_ERR0_SEC_IMG_VER_FAIL;
+			boot_err = BIT_ULL(CPU_BOOT_ERR_SEC_IMG_VER_FAIL);
 			msg = "Secondary";
 		}
-		cpu_boot_err_clrset(bit, true);
+		cpu_boot_err_clrset(boot_err, true);
 		preboot_puts_raw(msg);
 		preboot_puts(" is not a valid image.\n");
 	}
@@ -207,7 +208,7 @@
 	bool is_debug_en;
 
 	is_debug_en = hl_is_jtag_enabled();
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_DEBUG_EN, is_debug_en);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_DEBUG_EN), is_debug_en);
 	reg = regmap_readl(mmCPU_CA53_CFG_ARM_RST_CONTROL);
 	if (is_debug_en)
 		reg |= BIT(12);
@@ -249,7 +250,7 @@
 	if (__stack != STACK_CANARY_MAGIC_WORD) {
 		preboot_puts("The stack has overflowed\n");
 		preboot_puts_hex("Stack Canary value = ", __stack, false);
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_BOOT_FW_CRIT_ERR, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_BOOT_FW_CRIT_ERR), true);
 	}
 
 	ppboot_check_stack_mon_dev_specific();
diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index b53ea7d..143349c 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -322,12 +322,12 @@
 		case HL_FW_FIT_BOOT_ERR_CORRUPTED:
 		case HL_FW_FIT_BOOT_ERR_ADDRESS_VIOLATION:
 		case HL_FW_FIT_BOOT_ERR_NOT_RUNNABLE:
-			cpu_boot_err_clrset(CPU_BOOT_ERR0_FIT_CORRUPTED, true);
+			cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_FIT_CORRUPTED), true);
 			break;
 		case HL_FW_FIT_BOOT_ERR_REVOKED_KEY:
 		case HL_FW_FIT_BOOT_ERR_SVN:
 		case HL_FW_FIT_BOOT_ERR_SIGNATURE:
-			cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_FAIL, true);
+			cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_SECURITY_FAIL), true);
 			break;
 		}
 
@@ -393,9 +393,9 @@
 	cpu_boot_if_init();
 	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_PREBOOT);
 	/* Enable the errors communication */
-	cpu_boot_err_clrset(CPU_BOOT_ERR0_ENABLED, true);
+	cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_ENABLED), true);
 	/* Enable the device status communication */
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_ENABLED, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_ENABLED), true);
 
 	/*
 	 * Note: we assume CPU frequency is not changed while CPU is running,
diff --git a/arch/gaudi/arm/asic.c b/arch/gaudi/arm/asic.c
index 1c08fd2..e191143 100644
--- a/arch/gaudi/arm/asic.c
+++ b/arch/gaudi/arm/asic.c
@@ -249,7 +249,7 @@
 		base += TPC_QMAN_OFFSET;
 	}
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_CLK_GATE_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_CLK_GATE_EN), true);
 }
 
 void hl_ram_init(void)
@@ -533,7 +533,7 @@
 		regmap_writel(1, base + mmDMA_IF_W_S_DOWN_CH1_RL_SRAM_EN);
 	}
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_RL_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_RL_EN), true);
 }
 
 #define HBM0_WR_VAL		0x33333333
@@ -579,7 +579,7 @@
 		regmap_writel(mask, base + mmDMA_IF_W_S_HBM_CRED_EN_1);
 	}
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_HBM_CRED_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_HBM_CRED_EN), true);
 }
 
 void preboot_e2e_credits_init(void)
@@ -730,7 +730,7 @@
 	regmap_writel(BIT(0), mmDMA_IF_W_S_DOWN_CH1_E2E_HBM_EN);
 	regmap_writel(BIT(0), mmDMA_IF_W_S_DOWN_CH1_E2E_PCI_EN);
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_E2E_CRED_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_E2E_CRED_EN), true);
 }
 
 #define TPC_PREFETCH_HMASK	0x7F
diff --git a/arch/gaudi/arm/common.c b/arch/gaudi/arm/common.c
index 08c84e3..c484429 100644
--- a/arch/gaudi/arm/common.c
+++ b/arch/gaudi/arm/common.c
@@ -412,7 +412,7 @@
 
 void hl_plat_early_init(void)
 {
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_IATU_CONF_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN), true);
 }
 
 
@@ -450,7 +450,7 @@
 
 void lkd_dev_specific_setup(void)
 {
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_FW_HARD_RST_EN), true);
 }
 
 /* Hard coded each of the gaudi on-chip TS device properties */
@@ -497,7 +497,7 @@
 		regmap_writel(cold_rst_data.data, mmCOLD_RST_DATA);
 	}
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_SP_SRAM_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_SP_SRAM_EN), true);
 }
 
 void preboot_dev_enable_caps(void)
@@ -609,5 +609,5 @@
 void ppboot_check_stack_mon_dev_specific(void)
 {
 	if (__stack2 != STACK_CANARY_MAGIC_WORD)
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_BOOT_FW_CRIT_ERR, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_BOOT_FW_CRIT_ERR), true);
 }
diff --git a/arch/gaudi/arm/rma_info.c b/arch/gaudi/arm/rma_info.c
index 0598267..76fba4a 100644
--- a/arch/gaudi/arm/rma_info.c
+++ b/arch/gaudi/arm/rma_info.c
@@ -208,7 +208,7 @@
 		err_sts_reg_val |= HL_ERROR_STATUS_FATAL_ERROR;
 		i2csi_update_smbus_alert(rma_header);
 		/* set boot error bit for RMA */
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_DEVICE_UNUSABLE_FAIL, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_DEVICE_UNUSABLE_FAIL), true);
 		preboot_puts_raw("RMA Fatal error(s): ");
 		preboot_rma_type_print(rma_header->rma_failure);
 		i2c_rma_failure = rma_header->rma_failure;
diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 33b596d..c5e7045 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -1872,7 +1872,7 @@
 {
 	bool is_sec_en;
 
-	cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_NOT_RDY, true);
+	cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_SECURITY_NOT_RDY), true);
 	preboot_puts("Security init started...\n");
 
 	is_sec_en = preboot_security_is_enabled();
@@ -1902,9 +1902,9 @@
 	/* Store security info to be used by the booted image */
 	preboot_security_sp_sram_init();
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_SECURITY_EN, is_sec_en);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_SECURITY_EN), is_sec_en);
 
-	cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_NOT_RDY, false);
+	cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_SECURITY_NOT_RDY), false);
 	preboot_puts("Security init finished.\n");
 }
 
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index e083d44..74c26a1 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -116,12 +116,12 @@
 		regmap_writel(cold_rst_data.data, mmCOLD_RST_DATA);
 	}
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_SP_SRAM_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_SP_SRAM_EN), true);
 }
 
 void hl_plat_early_init(void)
 {
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_IATU_CONF_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN), true);
 
 	gaudi2_axi_drain_ena();
 
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index eb01936..486c205 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -586,7 +586,7 @@
 	u32 retry = ARC_HALT_ACK_RETRIES;
 	u32 halt_mask, halt_ack_mask;
 	u32 rst_offset;
-	u32 cpu_boot_err_val;
+	u64 cpu_boot_err_val;
 
 	if (id) {
 		halt_req_offset = mmPSOC_ARC1_CFG_RUN_HALT_REQ;
@@ -617,8 +617,8 @@
 			if (!retry) {
 				preboot_puts_hex("No ack from ARC", id, false);
 				cpu_boot_err_val = (!id) ?
-					CPU_BOOT_ERR0_ARC0_HALT_ACK_NOT_RCVD :
-					CPU_BOOT_ERR0_ARC1_HALT_ACK_NOT_RCVD;
+					BIT_ULL(CPU_BOOT_ERR_ARC0_HALT_ACK_NOT_RCVD) :
+					BIT_ULL(CPU_BOOT_ERR_ARC1_HALT_ACK_NOT_RCVD);
 				cpu_boot_err_clrset(cpu_boot_err_val, true);
 				break;
 			}
@@ -641,7 +641,7 @@
 	u32 run_ack_offset, run_req_offset;
 	u32 run_ack_mask;
 	u64 start;
-	u32 cpu_boot_err_val;
+	u64 cpu_boot_err_val;
 
 	if (id) {
 		run_ack_offset = mmPSOC_ARC1_CFG_RUN_HALT_ACK;
@@ -665,8 +665,8 @@
 			preboot_puts_hex("No running state ack from ARC ",
 					id, false);
 			cpu_boot_err_val = (!id) ?
-				CPU_BOOT_ERR0_ARC0_RUN_ACK_NOT_RCVD :
-				CPU_BOOT_ERR0_ARC1_RUN_ACK_NOT_RCVD;
+				BIT_ULL(CPU_BOOT_ERR_ARC0_RUN_ACK_NOT_RCVD) :
+				BIT_ULL(CPU_BOOT_ERR_ARC1_RUN_ACK_NOT_RCVD);
 			cpu_boot_err_clrset(cpu_boot_err_val, true);
 			break;
 		}
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 4145d26..c70e95d 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -144,7 +144,7 @@
 
 	init_secondary();
 
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_IATU_CONF_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN), true);
 
 	/*
 	 * Upon MANUAL, WATCHDOG, SW ALL, PRSTN, FLR and FW reset
diff --git a/arch/greco/arm/common.c b/arch/greco/arm/common.c
index 6fe5261..a751e45 100644
--- a/arch/greco/arm/common.c
+++ b/arch/greco/arm/common.c
@@ -143,7 +143,7 @@
 {
 	axi_enable_drain();
 	axi_enable_pcie_drain();
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_IATU_CONF_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN), true);
 }
 
 void ppboot_log_waiting_for_pcie(void)
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index fb5faf5..0ec096b 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -159,7 +159,7 @@
 	u32 run_ack_offset, run_req_offset;
 	u32 run_ack_mask;
 	u64 start;
-	u32 cpu_boot_err_val;
+	u64 cpu_boot_err_val;
 
 	if (id) {
 		run_ack_offset = mmPSOC_ARC1_CFG_RUN_HALT_ACK;
@@ -183,8 +183,8 @@
 			preboot_puts_hex("No running state ack from ARC ",
 					id, false);
 			cpu_boot_err_val = (!id) ?
-				CPU_BOOT_ERR0_ARC0_RUN_ACK_NOT_RCVD :
-				CPU_BOOT_ERR0_ARC1_RUN_ACK_NOT_RCVD;
+				BIT_ULL(CPU_BOOT_ERR_ARC0_RUN_ACK_NOT_RCVD) :
+				BIT_ULL(CPU_BOOT_ERR_ARC1_RUN_ACK_NOT_RCVD);
 			cpu_boot_err_clrset(cpu_boot_err_val, true);
 			break;
 		}
@@ -294,7 +294,7 @@
 
 void lkd_dev_specific_setup(void)
 {
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_FW_HARD_RST_EN), true);
 }
 
 void preboot_disable_pcie_ltr(void)
@@ -513,7 +513,7 @@
 	u32 retry = ARC_HALT_ACK_RETRIES;
 	u32 halt_mask, halt_ack_mask;
 	u32 rst_offset;
-	u32 cpu_boot_err_val;
+	u64 cpu_boot_err_val;
 
 	if (id) {
 		halt_ack_offset = mmPSOC_ARC1_CFG_RUN_HALT_ACK;
@@ -545,8 +545,8 @@
 			if (!retry) {
 				preboot_puts_hex("No ack from ARC", id, false);
 				cpu_boot_err_val = (!id) ?
-					CPU_BOOT_ERR0_ARC0_HALT_ACK_NOT_RCVD :
-					CPU_BOOT_ERR0_ARC1_HALT_ACK_NOT_RCVD;
+					BIT_ULL(CPU_BOOT_ERR_ARC0_HALT_ACK_NOT_RCVD) :
+					BIT_ULL(CPU_BOOT_ERR_ARC1_HALT_ACK_NOT_RCVD);
 				cpu_boot_err_clrset(cpu_boot_err_val, true);
 				break;
 			}
diff --git a/include/boot_if.h b/include/boot_if.h
index 935c7e3..5003ad9 100644
--- a/include/boot_if.h
+++ b/include/boot_if.h
@@ -16,9 +16,9 @@
 u32 cpu_boot_cmd_get(void);
 void cpu_boot_cmd_clr_before_wfe(void);
 void cpu_boot_sts_set(u32 val);
-void cpu_boot_err_clrset(u32 mask, bool set);
-bool cpu_boot_err_get(u32 mask);
-void cpu_boot_dev_sts_clrset(u32 mask, bool set);
+void cpu_boot_err_clrset(u64 mask, bool set);
+bool cpu_boot_err_get(u64 mask);
+void cpu_boot_dev_sts_clrset(u64 mask, bool set);
 void cpu_boot_rsp_set(enum cpu_msg_status sts);
 u64 cpu_boot_dev_sts_get_all(void);
 void cpu_boot_if_fix_ro_regs(void);
diff --git a/lib/common/boot_if.c b/lib/common/boot_if.c
index bfc8866..f969a16 100644
--- a/lib/common/boot_if.c
+++ b/lib/common/boot_if.c
@@ -12,23 +12,22 @@
 #include <preboot.h>
 #include <hl_boot_if.h>			/* requires stdint.h */
 
-/* TODO: add support for both registers of status and error */
 struct cpu_boot_if {
 	u32 status;	/* boot progress status register offset */
 	u32 response;
-	u32 err;
-	u32 dev_sts;
+	u32 err[2];
+	u32 dev_sts[2];
 	/* Shadow the status and error registers */
-	u32 dev_sts_val;
-	u32 err_val;
+	u32 dev_sts_val[2];
+	u32 err_val[2];
 	u32 status_val;	/* boot progress status value */
 };
 
 static struct cpu_boot_if cpu_boot_if = {
 	.status = mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS,
 	.response = mmCPU_CMD_STATUS_TO_HOST,
-	.err = mmCPU_BOOT_ERR0,
-	.dev_sts = mmCPU_BOOT_DEV_STS0,
+	.err = { mmCPU_BOOT_ERR0, mmCPU_BOOT_ERR1 },
+	.dev_sts = { mmCPU_BOOT_DEV_STS0, mmCPU_BOOT_DEV_STS1 },
 };
 
 /*
@@ -59,14 +58,18 @@
 
 void cpu_boot_if_init(void)
 {
-	cpu_boot_if.dev_sts_val = regmap_readl(cpu_boot_if.dev_sts);
-	cpu_boot_if.err_val = regmap_readl(cpu_boot_if.err);
+	cpu_boot_if.dev_sts_val[0] = regmap_readl(cpu_boot_if.dev_sts[0]);
+	cpu_boot_if.dev_sts_val[1] = regmap_readl(cpu_boot_if.dev_sts[1]);
+	cpu_boot_if.err_val[0] = regmap_readl(cpu_boot_if.err[0]);
+	cpu_boot_if.err_val[1] = regmap_readl(cpu_boot_if.err[1]);
 }
 
 void cpu_boot_if_fix_ro_regs(void)
 {
-	regmap_writel(cpu_boot_if.dev_sts_val, cpu_boot_if.dev_sts);
-	regmap_writel(cpu_boot_if.err_val, cpu_boot_if.err);
+	regmap_writel(cpu_boot_if.dev_sts_val[0], cpu_boot_if.dev_sts[0]);
+	regmap_writel(cpu_boot_if.dev_sts_val[1], cpu_boot_if.dev_sts[1]);
+	regmap_writel(cpu_boot_if.err_val[0], cpu_boot_if.err[0]);
+	regmap_writel(cpu_boot_if.err_val[1], cpu_boot_if.err[1]);
 	regmap_writel(cpu_boot_if.status_val, cpu_boot_if.status);
 }
 
@@ -74,10 +77,14 @@
 {
 	u32 *shadow;
 
-	if (reg == cpu_boot_if.dev_sts)
-		shadow = &cpu_boot_if.dev_sts_val;
-	else if (reg == cpu_boot_if.err)
-		shadow = &cpu_boot_if.err_val;
+	if (reg == cpu_boot_if.dev_sts[0])
+		shadow = &cpu_boot_if.dev_sts_val[0];
+	else if (reg == cpu_boot_if.dev_sts[1])
+		shadow = &cpu_boot_if.dev_sts_val[1];
+	else if (reg == cpu_boot_if.err[0])
+		shadow = &cpu_boot_if.err_val[0];
+	else if (reg == cpu_boot_if.err[1])
+		shadow = &cpu_boot_if.err_val[1];
 	else
 		return;
 
@@ -88,21 +95,22 @@
 	regmap_writel(*shadow, reg);
 }
 
-/* We support only ERR0 for now */
-void cpu_boot_err_clrset(u32 mask, bool set)
+void cpu_boot_err_clrset(u64 mask, bool set)
 {
-	cpu_boot_reg_clrset(cpu_boot_if.err, mask, set);
+	cpu_boot_reg_clrset(cpu_boot_if.err[0], (u32)mask, set);
+	cpu_boot_reg_clrset(cpu_boot_if.err[1], (u32)(mask >> 32), set);
 }
 
-bool cpu_boot_err_get(u32 mask)
+bool cpu_boot_err_get(u64 mask)
 {
-	return !!(cpu_boot_if.err_val & mask);
+	return !!((cpu_boot_if.err_val[0] & (u32)mask) |
+		  (cpu_boot_if.err_val[1] & (u32)(mask >> 32)));
 }
 
-/* We support only STS0 for now */
-void cpu_boot_dev_sts_clrset(u32 mask, bool set)
+void cpu_boot_dev_sts_clrset(u64 mask, bool set)
 {
-	cpu_boot_reg_clrset(cpu_boot_if.dev_sts, mask, set);
+	cpu_boot_reg_clrset(cpu_boot_if.dev_sts[0], (u32)mask, set);
+	cpu_boot_reg_clrset(cpu_boot_if.dev_sts[1], (u32)(mask >> 32), set);
 }
 
 void cpu_boot_rsp_set(enum cpu_msg_status sts)
@@ -117,7 +125,7 @@
 
 u64 cpu_boot_dev_sts_get_all(void)
 {
-	u64 val = cpu_boot_if.dev_sts_val;
+	u64 val = cpu_boot_if.dev_sts_val[1];
 
-	return val;
+	return val << 32 | cpu_boot_if.dev_sts_val[0];
 }
diff --git a/lib/common/efuse.c b/lib/common/efuse.c
index b2088ff..be9f900 100644
--- a/lib/common/efuse.c
+++ b/lib/common/efuse.c
@@ -59,14 +59,14 @@
 	u32 read_val;
 
 	if (word >= FUSE_WORDS_PER_BANK) {
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_EFUSE_FAIL, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_EFUSE_FAIL), true);
 		preboot_puts("Invalid efuse word offset\n");
 		return -1;
 	}
 
 	ret = preboot_fuse_init();
 	if (ret == -2) {
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_EFUSE_FAIL, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_EFUSE_FAIL), true);
 		preboot_puts("Efuse not initialized\n");
 		return ret;
 	}
diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 0fcaae0..49855cc 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -7,6 +7,7 @@
 
 #include <libfdt.h>
 #include <string.h>
+#include <bitops.h>
 #include <security.h>
 #include <hl_boot_if.h>
 #include <hl_boot_desc.h>
@@ -962,8 +963,8 @@
 	if (!err)
 		err = fit_copy_images(fit, &sec_res);
 
-	cpu_boot_err_clrset(CPU_BOOT_ERR0_FIT_CORRUPTED |
-			    CPU_BOOT_ERR0_SECURITY_FAIL, false);
+	cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_FIT_CORRUPTED) |
+			    BIT_ULL(CPU_BOOT_ERR_SECURITY_FAIL), false);
 	switch (err) {
 	case FIT_LOAD_SUCCESS:
 		return 0;
@@ -971,12 +972,12 @@
 	case FIT_LOAD_BAD_LOAD_ADDR:
 	case FIT_LOAD_NOT_RUNNABLE:
 	case FIT_LOAD_CORRUPTED:
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_FIT_CORRUPTED, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_FIT_CORRUPTED), true);
 		preboot_error_inform(HL_FW_ERR_FIT, err);
 		preboot_puts("FIT loading failed, Retrying...\n");
 		break;
 	case FIT_LOAD_SEC_FAILED:
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_FAIL, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_SECURITY_FAIL), true);
 		preboot_error_inform(HL_FW_ERR_SECURITY,
 				     fwupd_sec_convert_min_error(sec_res));
 		preboot_puts("FIT security validation failed, Retrying...\n");
diff --git a/lib/preboot/memory.c b/lib/preboot/memory.c
index 07bc63f..38f44b7 100644
--- a/lib/preboot/memory.c
+++ b/lib/preboot/memory.c
@@ -6,6 +6,7 @@
  */
 
 #include <preboot.h>
+#include <bitops.h>
 #include <memory.h>
 #include <boot_if.h>
 
@@ -73,7 +74,7 @@
 {
 	bool retval = false;
 
-	if (cpu_boot_err_get(CPU_BOOT_ERR0_BOOT_FW_CRIT_ERR)) {
+	if (cpu_boot_err_get(BIT_ULL(CPU_BOOT_ERR_BOOT_FW_CRIT_ERR))) {
 		preboot_puts("Critical firmware error occurred\n");
 		preboot_puts("Please fix the code to avoid that\n");
 		retval = true;
@@ -83,7 +84,7 @@
 		preboot_puts("Preboot stack has oveflowed and killed the poor canary :(\n");
 		preboot_puts_hex("Canary value = ", __stack, false);
 		preboot_puts("Please fix the code to avoid stack overflow.\n");
-		cpu_boot_err_clrset(CPU_BOOT_ERR0_BOOT_FW_CRIT_ERR, true);
+		cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_BOOT_FW_CRIT_ERR), true);
 		retval = true;
 	}
 
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 5c10f9d..700e03c 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -10,6 +10,7 @@
 #include <preboot.h>
 #include <boot_if.h>
 #include <boot_info.h>
+#include <bitops.h>
 #include <security.h>
 #include <crc32.h>
 #include <memory.h>
@@ -37,7 +38,7 @@
 
 void module_lkd_enable(void)
 {
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
+	cpu_boot_dev_sts_clrset(BIT_ULL(CPU_BOOT_DEV_STS_FW_LD_COM_EN), true);
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 }
 
@@ -605,8 +606,8 @@
 	 * List of status bits we clear because we do not know if the FW we are
 	 * loading supports these functionalities
 	 */
-	u32 sts_msk = CPU_BOOT_DEV_STS0_FW_HARD_RST_EN |
-		CPU_BOOT_DEV_STS0_FW_LD_COM_EN;
+	u64 sts_msk = BIT_ULL(CPU_BOOT_DEV_STS_FW_HARD_RST_EN) |
+		      BIT_ULL(CPU_BOOT_DEV_STS_FW_LD_COM_EN);
 
 	preboot_security_cleanup_before_run();
 	hl_ts_shutdown();
